src/Raven.Server/Documents/Indexes/Static/JavaScriptIndex.cs

```diff
@@ -223,7 +224,19 @@ private JsValue LoadDocument(JsValue self, JsValue[] args)
         reduce.aggregateBy = reduceFunction;
     }
     return reduce;
-}";
+}
+
+function createSpatialPoint(lat, lng) {
```

Comment:
I don't like this:
`createSpatialField(createSpatialPoint(l.Latitude, l.Longitude))}`

Let's do "overloading" here by checking number of args and their types.
Using `arguments`, we have the following signatures:

* `createSpatialField(name, lat,lng)`
* `createSpatialField(name, wkt)`
* `createSpatialField(lat, lng)`
* `createSpatialField(wkt)`

We can use the types to decide how to deal with this directly

Reply:
No overloading in JS, the 'createSpatialPoint' is just an helper function you can invoke it like so: 
createSpatialField({$spatial:{Lat:x.lat, Lng:x.lng}})      // (assuming i'll remove the Type parameters)

Reply:
No, overloading won't work in JS, but _varaible number of args_ work

Reply:
I have looked at the JS community and asked and this was the answer I got, I don't claim to know what makes sense in JS. I'll check the number of arguments, I don't really have an opinion about this...

Reply:
@talweiss1982 maybe this will be useful for you:
https://www.typescriptlang.org/docs/handbook/functions.html#overloads

Ex.

function pickCard(x: {suit: string; card: number; }[]): number;
function pickCard(x: number): {suit: string; card: number; };
function pickCard(x): any {
    // Check to see if we're working with an object/array
    // if so, they gave us the deck and we'll pick the card
    if (typeof x == "object") {
        let pickedCard = Math.floor(Math.random() * x.length);
        return pickedCard;
    }
    // Otherwise just let them pick the card
    else if (typeof x == "number") {
        let pickedSuit = Math.floor(x / 13);
        return { suit: suits[pickedSuit], card: x % 13 };
    }
}